exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -68,5 +79,151 @@ func (prwe *prwExporter) shutdown(context.Context) error {
 // TimeSeries, validates and handles each individual metric, adding the converted TimeSeries to the map, and finally
 // exports the map.
 func (prwe *prwExporter) pushMetrics(ctx context.Context, md pdata.Metrics) (int, error) {
-	return 0, nil
+	prwe.wg.Add(1)
+	defer prwe.wg.Done()
+	select {
+	case <-prwe.closeChan:
+		return pdatautil.MetricCount(md), errors.New("shutdown has been called")
+	default:
+		tsMap := map[string]*prompb.TimeSeries{}
+		dropped := 0
+		errs := []string{}
+
+		resourceMetrics := data.MetricDataToOtlp(pdatautil.MetricsToInternalMetrics(md))
+		for _, resourceMetric := range resourceMetrics {
+			// TODO: add resource attributes as labels, probably in next PR
+			for _, instrumentationMetrics := range resourceMetric.InstrumentationLibraryMetrics {
+				// TODO: decide if instrumentation library information should be exported as labels
+				for _, metric := range instrumentationMetrics.Metrics {
+					// check for valid type and temporality combination
+					if ok := validateMetrics(metric.MetricDescriptor); !ok {
+						dropped++
+						errs = append(errs, "invalid temporality and type combination")
+						continue
+					}
+					// handle individual metric based on type
+					switch metric.GetMetricDescriptor().GetType() {
+					case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64,
+						otlp.MetricDescriptor_MONOTONIC_DOUBLE, otlp.MetricDescriptor_DOUBLE:
+						if err := prwe.handleScalarMetric(tsMap, metric); err != nil {
+							dropped++
+							errs = append(errs, err.Error())
+						}
+					}
+				}
+			}
+		}
+
+		if err := prwe.export(ctx, tsMap); err != nil {
+			return pdatautil.MetricCount(md), err
+		}
+
+		if dropped != 0 {
+			return dropped, errors.New(strings.Join(errs, "\n"))
```

Comment:
nit: I wonder if there's an established pattern for this sort of thing--maybe something like `hashicorp/go-multierror`? The reason I ask is that it's an expensive string to join before knowing whether it will be used.

Reply:
You are right. I did find a componenterror.CombineErrors() method that combines error. Thanks.